tools/automation/building_blocks/floor.py

```diff
@@ -0,0 +1,57 @@
+"""
+  Contains classes which are used to store data for a floor.
+
+  Ready to use, or extend subclasses as required to fill cases where more 
+  information needs to be stored.
+
+  Call to_dictionary() to retrieve all the data.
+"""
+
+import abc
+
+class Floor:
+  
+  def __init__(self, floor_name, floor_id):
+    self._name = floor_name
+    self._id = "FACILITIES/{0}".format(floor_id)
+    self._connections = {}
+  
+  def populate_connections(self, building_name):
+    self._connections.update({building_name: "CONTAINS"})

```

Comment:
nit: for single key updates maybe prefer assignment

`self._connections[building_name] = "CONTAINS"`

more readable and slightly better performance

https://stackoverflow.com/a/15456218